### ISSUE TITLE
How to pass some object / parameter into custom RssReader / AbstractRssReader?

### ISSUE THREAD

**Issue by aswzen** [w3stling/rssreader#58]
I want to do some logic in the registerItemTags() based on several conditions,

For example when creating the reader, i passed some param thru the header,

`List<Item> items = new RssReader().addHeader("Param","1").read("http://google.com").toList();`

Then i want to read it again,

```
public class MovieRssReader extends AbstractRssReader<Channel, Item> {

    @Override
    protected void registerItemTags() {
           if(parent.getHeader("Param") == "1")
                do this
    }
```

Any help will be appreciated, thanks

**Comment by aswzen** [comment]
Already tried to make some variable setter on the custom reader class like

```
List<Item> items = new MovieRssReader()
                            .setSource("Genflix")
                            .read(rssSource.getRssFeedLink()).toList();
```

and in the reader like
```
public class CustomRssReader extends AbstractRssReader<Channel, Item> {

    public String source;

 @Override
    protected void registerItemTags() {
        super.registerItemTags();
        
        if(this.source.equalsIgnoreCase("Genflix")){
               //do something
        }

```

but this.source still null

**Comment by aswzen** [comment]
The objective is simple, I can set something into the parent class, and also can get it again while doing registerItemTags().

Now its working by adding new variable inside the abstraction (maybe you can add this feature in the future), plus with their getter and setter, 

![image](https://user-images.githubusercontent.com/1474916/211076606-9d58ac5f-ffb8-483b-bcc9-2459b7ade7c0.png)

**Comment by w3stling** [comment]
This is now fixed by PR #59 and is it included in [3.4.0](https://search.maven.org/artifact/com.apptasticsoftware/rssreader/3.4.0/jar) release.

`setSource("Genflix")` should work from version 3.4.0.
I would avoid passing parameters via `addHeader("Param","1")` as this will set the parameter to the http request header.

**Comment by aswzen** [comment]
Ok thanks, really appreciated it
